x-pack/legacy/plugins/siem/public/pages/detection_engine/create_rule/components/description_step/index.tsx

```diff
@@ -0,0 +1,137 @@
+  ({ data, indexPatterns, schema }) => {
+    const keys = Object.keys(schema);
+    return (
+      <EuiFlexGroup gutterSize="none" direction="row" justifyContent="spaceAround">
+        {chunk(keys.includes('queryBar') ? 3 : Math.ceil(keys.length / 2), keys).map(key => (
+          <EuiFlexItem grow={false}>
+            <EuiDescriptionList
+              listItems={buildListItems(data, pick(key, schema), indexPatterns)}
+            />
+          </EuiFlexItem>
+        ))}
+      </EuiFlexGroup>
+    );
+  }
+);
+
+interface ListItems {
+  title: NonNullable<ReactNode>;
+  description: NonNullable<ReactNode>;
+}
+
+const buildListItems = (
+  data: unknown,
+  schema: FormSchema,
+  indexPatterns?: IIndexPattern
+): ListItems[] =>
+  Object.keys(schema).reduce<ListItems[]>(
+    (acc, field) => [
+      ...acc,
+      ...getDescriptionItem(field, get([field, 'label'], schema), data, indexPatterns),
+    ],
+    []
+  );
+
+const getDescriptionItem = (
+  field: string,
+  label: string,
+  value: unknown,
+  indexPatterns?: IIndexPattern
+): ListItems[] => {
+  if (field === 'queryBar' && indexPatterns != null) {
+    const filters = get('queryBar.filters', value) as esFilters.Filter[];
+    const query = get('queryBar.query', value) as Query;
+    const savedId = get('queryBar.saved_id', value);
+    let items: ListItems[] = [];
+    if (!isEmpty(filters)) {
+      items = [
+        ...items,
+        {
+          title: <>{I18n.FILTERS_LABEL}</>,
+          description: (
+            <EuiFlexGroup wrap responsive={false} gutterSize="xs">
+              {filters.map((filter, index) => (
+                <EuiFlexItem grow={false} key={`${field}-filter-${index}`}>
+                  <EuiBadge color="hollow">
+                    <FilterLabel
+                      filter={filter}
+                      valueLabel={utils.getDisplayValueFromFilter(filter, [indexPatterns])}
+                    />
+                  </EuiBadge>
+                </EuiFlexItem>
+              ))}
+            </EuiFlexGroup>
+          ),
+        },
+      ];
+    }
+    if (!isEmpty(query.query)) {
+      items = [
+        ...items,
+        {
+          title: <>{I18n.QUERY_LABEL}</>,
+          description: <>{query.query}</>,
+        },
+      ];
+    }
+    if (!isEmpty(savedId)) {
+      items = [
+        ...items,
+        {
+          title: <>{I18n.SAVED_ID_LABEL}</>,
+          description: <>{savedId}</>,
+        },
+      ];
+    }
+    return items;
+  } else if (Array.isArray(value)) {
+    return [
+      {
+        title: label,
+        description: (
+          <EuiFlexGroup wrap responsive={false} gutterSize="xs">
+            {value.map(val => (
+              <EuiFlexItem grow={false} key={`${field}-${val}`}>
+                <EuiBadge color="hollow">{val}</EuiBadge>
+              </EuiFlexItem>
+            ))}
+          </EuiFlexGroup>
+        ),
+      },
+    ];
+  }
```

Comment:
Not sure this is functioning for `Reference URL`'s and `False Positives (as they are just concatenated without a space)

Edit
<img src="https://user-images.githubusercontent.com/2946766/69586894-36b1e180-0fa1-11ea-9dc2-c476cf9d382a.png" width=400px> 

Preview
<img src="https://user-images.githubusercontent.com/2946766/69586922-4a5d4800-0fa1-11ea-9e34-9c70e32aaae1.png" width=400px>